knimin/templates/pm_extract_plate.html

```diff
@@ -0,0 +1,186 @@
+{% extends logged_in_index.html %}
+{% block head %}
+<script>
+  /**
+   *
+   * Submits the selected plates for extraction
+   *
+   **/
+  function submitPlateExtraction() {
+    // Clean up warnings
+    $("#robot-warning").hide();
+    $("#tool-warning").hide();
+    $("#kit-warning").hide();
+
+    // Retrieve information
+    var plates = [];
+    $.each($(".list-group-item"), function(idx, item) {
+      plates.push($(item).attr("pm-id"));
+    });
+
+    var robot = $("#extraction-robot-select").val();
+    var tool = $("#extraction-tool-select").val();
+    var kit = $("#extraction-kit-input").val().trim();
+
+    // Validation
+    var success = true;
+    if (robot === "-1") {
+      success = false;
+      $("#robot-warning").show();
+    }
+    if (tool === "-1") {
+      success = false;
+      $("#tool-warning").show();
+    }
+    if (kit.length < 1) {
+      success = false;
+      $("#kit-warning").show();
+    }
+
+    if (success) {
+      var form = $('<form>', {action: $(location).attr('href'), method: 'post'});
+      var fields = {'plates': JSON.stringify(plates), 'robot': robot, 'tool': tool, 'kit': kit};
+      $.each(fields, function(key, val) {
+        $('<input>').attr({type: "hidden", name: key, value: val}).appendTo(form);
+      });
+      form.appendTo('body').submit();
+    }
+  };
+
+  /**
+   *
+   * Formats the background color of the kit lot select
+   *
+   **/
+  function formatKitLot(kits) {
+    var val = $("#extraction-kit-input").val().trim();
+    $("#extraction-kit-input").removeClass('pm-warning');
+    if (val.length > 0 && $.inArray(val, kits) === -1) {
+      $("#extraction-kit-input").addClass('pm-warning');
+    }
+  }
+
+  $(document).ready(function() {
+    var plates = {% raw [ [int(p[0]), p[1]] for p in plates ] %};
```

Comment:
Instead of formatting the list, perhaps it would be more convenient to make sure the data is in a JSON-compatible format and then just call json.dumps.